client/azure_client_set.go

```
@@ -63,117 +65,185 @@ func NewAzureClientSet(config AzureClientSetConfig) (*AzureClientSet, error) {
-		StorageAccountsClient:                  newStorageAccountsClient(c),
-		UsageClient:                            newUsageClient(c),
-		VirtualNetworkClient:                   newVirtualNetworkClient(c),
-		VirtualNetworkGatewayConnectionsClient: newVirtualNetworkGatewayConnectionsClient(c),
-		VirtualNetworkGatewaysClient:           newVirtualNetworkGatewaysClient(c),
-		VirtualMachineScaleSetVMsClient:        newVirtualMachineScaleSetVMsClient(c),
-		VirtualMachineScaleSetsClient:          newVirtualMachineScaleSetsClient(c),
-		VnetPeeringClient:                      newVnetPeeringClient(c),
+		DeploymentsClient:                      deploymentsClient,
+		DNSRecordSetsClient:                    dnsRecordSetsClient,
+		DNSZonesClient:                         dnsZonesClient,
+		GroupsClient:                           groupsClient,
+		InterfacesClient:                       interfacesClient,
+		StorageAccountsClient:                  storageAccountsClient,
+		UsageClient:                            usageClient,
+		VirtualNetworkClient:                   virtualNetworkClient,
+		VirtualNetworkGatewayConnectionsClient: virtualNetworkGatewayConnectionsClient,
+		VirtualNetworkGatewaysClient:           virtualNetworkGatewaysClient,
+		VirtualMachineScaleSetVMsClient:        virtualMachineScaleSetVMsClient,
+		VirtualMachineScaleSetsClient:          virtualMachineScaleSetsClient,
+		VnetPeeringClient:                      vnetPeeringClient,
 	}
 
 	return clientSet, nil
 }
 
-func newDeploymentsClient(config *clientConfig) *resources.DeploymentsClient {
+func newDeploymentsClient(config *clientConfig) (*resources.DeploymentsClient, error) {
 	c := resources.NewDeploymentsClientWithBaseURI(config.resourceManagerEndpoint, config.subscriptionID)
 	c.Authorizer = autorest.NewBearerAuthorizer(config.servicePrincipalToken)
+	err := c.AddToUserAgent(config.partnerIdUserAgent)
 
-	return &c
+	return &c, microerror.Mask(err)
```

There is a convention to return masked error only when it's not `nil`. One reason for that is better diffs if more erroring calls are added.

```suggestion
	if err != nil {
		return nil, microerror.Mask(err)
	}
	
	return &c, nil
```

---

PTALA 😉

---

Already approved LGTM 👍